kotlin-native/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/cexport/CAdapterGenerator.kt

```diff
@@ -51,11 +52,19 @@ private enum class Direction {
     C_TO_KOTLIN
 }
 
+private val hiddenFromObjCAnnotation = FqName("kotlin.native.HiddenFromObjC")
+
+private fun hasHiddenAnnotation(descriptor: Annotated): Boolean {
+    return descriptor.annotations.hasAnnotation(hiddenFromObjCAnnotation)
```

Comment:
Please add a comment explaining the purpose of this condition and the trade-off.

I mean, it checks for `HiddenFromObjC` while C export has nothing to do with Objective-C, which might be confusing for anyone reading this code.
It is also a bit of a hack, but we decided that it is acceptable. Still, worth mentioning.